src/main.rs

```diff
@@ -69,32 +70,34 @@ fn main() {
 
     // If you need to regenerate the serialised files, change to if (true)
     if false {
-      serialise_list_immutable_array_i8("subpurpose_purpose_lookup");
-      serialise_list("start_nodes");
-      serialise_list("init_travel_times");
-      serialise_GraphWalk();
-      serialise_GraphPT();
-      serialise_list_of_lists("node_values");
-      serialise_list_of_lists("travel_time_relationships");
+        serialise_list_immutable_array_i8("subpurpose_purpose_lookup");
+        serialise_list("start_nodes");
+        serialise_list("init_travel_times");
+        serialise_GraphWalk();
+        serialise_GraphPT();
+        serialise_list_of_lists("node_values");
+        serialise_list_of_lists("travel_time_relationships");
     }
 
     let now = Instant::now();
-    let start_nodes = read_serialised_vect32("start_nodes");
-    let init_travel_times = read_serialised_vect32("init_travel_times");
-    let graph_walk = read_GraphWalk();
-    let graph_pt = read_GraphPT();
+    let start_nodes: Vec<i32> = deserialize_bincoded_file("start_nodes");
```

Comment:
The compiler figures out what type parameter `T` to use for the function in a few ways. In this case, it's easiest to just declare what type the variable has.